src/apps/HeyGPT/HeyGPT/Activation/FileTypeActivationHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Windows.ApplicationModel.Activation;
+using Microsoft.Windows.AppLifecycle;
+using AndGPT.Core.Contracts.Services;
+using AndGPT.Core.Events;
+using AndGPT.Core.Models;
+
+namespace HeyGPT.App.Activation;
+
+public sealed class FileTypeActivationHandler : ActivationHandler<AppActivationArguments>
+{
+    private const string HandlerName = "File";
+
+    private readonly IEventAggregator _eventAggregator;
+
+    public FileTypeActivationHandler(IEventAggregator eventAggregator)
```

Comment:
We inject an EventAggregator to allow us to decouple this activator from anybody that wants to know about the file activations.   In the LetsChatViewModel there is a subscription to the event fired by this class.   It will use the file info and send any message prompts it finds.